libs/pages/application/src/lib/ui/page-settings-resources/page-settings-resources.tsx

```diff
@@ -0,0 +1,185 @@
+import { FormEventHandler } from 'react'
+import { Controller, useFormContext } from 'react-hook-form'
+import { MemorySizeEnum } from '@console/shared/enums'
+import { ApplicationEntity } from '@console/shared/interfaces'
+import {
+  BlockContent,
+  Button,
+  ButtonSize,
+  ButtonStyle,
+  HelpSection,
+  Icon,
+  IconAwesomeEnum,
+  InputSelect,
+  InputText,
+  Slider,
+  WarningBox,
+  WarningBoxEnum,
+} from '@console/shared/ui'
+import { convertCpuToVCpu } from '@console/shared/utils'
+
+export interface PageSettingsResourcesProps {
+  onSubmit: FormEventHandler<HTMLFormElement>
+  handleChangeMemoryUnit: () => void
+  memorySize: MemorySizeEnum
+  displayWarningCpu: boolean
+  application?: ApplicationEntity
+  loading?: boolean
+}
+
+export function PageSettingsResources(props: PageSettingsResourcesProps) {
+  const { onSubmit, loading, handleChangeMemoryUnit, application, memorySize, displayWarningCpu } = props
+  const { control, formState, watch } = useFormContext()
+
+  const pattern = {
+    value: /^[0-9]+$/,
+    message: 'Please enter a number.',
+  }
+
+  const maxMemoryBySize =
+    memorySize === MemorySizeEnum.GB ? (application?.maximum_memory || 0) / 1024 : application?.maximum_memory || 0
+
+  if (!application) return null
+
+  return (
+    <div className="flex flex-col justify-between w-full">
+      <div className="p-8 max-w-content-with-navigation-left">
+        <h2 className="h5 mb-8 text-text-700">Resources</h2>
+        <form onSubmit={onSubmit}>
+          <p className="text-text-500 text-xs mb-3">Adapt the application's consumption accordingly</p>
+          <BlockContent title="vCPU">
+            <p className="flex items-center text-text-600 mb-3 font-medium">
+              {watch('cpu')}
+              {displayWarningCpu && (
+                <Icon name={IconAwesomeEnum.TRIANGLE_EXCLAMATION} className="ml-1 text-error-500 text-sm" />
+              )}
+            </p>
+            <Controller
+              name="cpu"
+              control={control}
+              rules={{
+                max: (application?.cpu || 0) / 1000,
+              }}
+              render={({ field }) => (
+                <Slider
```

Comment:
As you can see in the video, the jump of the warning icon can be a little bit unpleasant. What do you think about setting in min-width to the number so that at least for all numbers until 10 we don't have a small jump of the warning?

https://user-images.githubusercontent.com/6163954/187417489-5b17285d-7216-4e95-a4c7-38a27e87b3c7.mp4

Reply:
Yes I see, I added it at the left of the number because with a min-width the margin right is strange!
It should be better !!

<img width="707" alt="Capture d’écran 2022-08-30 à 15 58 14" src="https://user-images.githubusercontent.com/533928/187456582-1f0b85a0-47ff-4c1b-af27-80698f032c12.png">